bridge_adaptivity/module/views.py

```diff
@@ -20,24 +20,74 @@
 from bridge_lti.outcomes import update_lms_grades
 from module import utils
 from module.forms import ActivityForm
-from module.mixins import CollectionIdToContextMixin, LtiSessionMixin
-from module.models import Activity, Collection, Log, Sequence, SequenceItem
+from module.mixins import CollectionIdToContextMixin, CollectionMixin, GroupEditFormMixin, LtiSessionMixin
+from module.models import Activity, Collection, CollectionGroup, Log, Sequence, SequenceItem
+
 
 log = logging.getLogger(__name__)
 
 
 @method_decorator(login_required, name='dispatch')
-class CollectionList(ListView):
+class GroupList(ListView):
+    model = CollectionGroup
+    context_object_name = 'groups'
+    ordering = ['id']
+
+    def get_queryset(self):
+        return self.model.objects.filter(owner=self.request.user)
+
+
+@method_decorator(login_required, name='dispatch')
+class GroupCreate(GroupEditFormMixin, CreateView):
+    model = CollectionGroup
+    slug_field = 'slug'
+    slug_url_kwarg = 'group_slug'
+    fields = [
+        'name', 'owner', 'collections', 'engine'
+    ]
+
+
+@method_decorator(login_required, name='dispatch')
+class GroupDetail(DetailView):
+    model = CollectionGroup
+    slug_field = 'slug'
+    slug_url_kwarg = 'group_slug'
+    context_object_name = 'group'
+
+    def get_queryset(self):
+        return CollectionGroup.objects.filter(owner=self.request.user)
+
+
+@method_decorator(login_required, name='dispatch')
+class GroupUpdate(GroupEditFormMixin, UpdateView):
+    model = CollectionGroup
+    slug_field = 'slug'
+    slug_url_kwarg = 'group_slug'
+    fields = [
+        'name', 'owner', 'collections', 'engine'
+    ]
+
+    context_object_name = 'group'
+
+    def get_success_url(self):
+        return reverse('module:group-detail', kwargs={'pk': self.kwargs.get('pk')})
+
+
+@method_decorator(login_required, name='dispatch')
+class CollectionList(CollectionMixin, ListView):
     model = Collection
     context_object_name = 'collections'
     ordering = ['id']
 
     def get_queryset(self):
-        return Collection.objects.filter(owner=self.request.user)
+        qs = Collection.objects.filter(owner=self.request.user)
```

Comment:
You've already moved this code to the mixin and it is odd here or I missed something.

Reply:
fixed. I don't know why git hub did not remove your comment from conversation, but this line is not present anymore in code.